### PR TITLE
Consolidate functions in complex.rs

### PR DESCRIPTION
**Pull request by sffc**
Address https://github.com/unicode-org/icu4x/pull/3267#discussion_r1159541999

### CODE REVIEW

File: experimental/segmenter/src/complex.rs [hunk 1]
```diff
@@ -121,42 +121,23 @@ impl Dictionary {
...
-    >(
-        provider: &D,
-        locale: Locale,
-    ) -> Result<DataPayload<UCharDictionaryBreakDataV1Marker>, DataError> {
-        provider
-            .load(DataRequest {
-                locale: &DataLocale::from(locale),
-                metadata: {
-                    let mut m = DataRequestMetadata::default();
-                    m.silent = true;
```
Comment (aethanyc) [inline experimental/segmenter/src/complex.rs:142]: Question: we have `m.silent = true` here, but the new `load_w_auto` doesn't set it. Is it intentional?

Comment (sffc) [reply experimental/segmenter/src/complex.rs:142]: I think yes! Merge problems. I'll fix.

**Comment by robertbastian** [comment]
This is probably handled by #3270

**Comment by sffc** [comment]
Obsolete

**Pull request closed without merging**